swarm/fuse/swarmfs_test.go

```diff
@@ -278,527 +286,555 @@ func (ta *testAPI) maxMounts(t *testing.T) {
 	if err == nil {
 		t.Fatalf("Error: Going beyond max mounts  %v", bzzHash6)
 	}
-
 }
 
 func (ta *testAPI) remount(t *testing.T) {
-	files := make(map[string]fileInfo)
-	files["1.txt"] = fileInfo{0700, 333, 444, getRandomBtes(10)}
-	uploadDir1, _ := ioutil.TempDir(os.TempDir(), "re-upload1")
-	bzzHash1 := createTestFilesAndUploadToSwarm(t, ta.api, files, uploadDir1)
-	testMountDir1, _ := ioutil.TempDir(os.TempDir(), "re-mount1")
-	swarmfs := mountDir(t, ta.api, files, bzzHash1, testMountDir1)
-	defer swarmfs.Stop()
-
-	uploadDir2, _ := ioutil.TempDir(os.TempDir(), "re-upload2")
-	bzzHash2 := createTestFilesAndUploadToSwarm(t, ta.api, files, uploadDir2)
-	testMountDir2, _ := ioutil.TempDir(os.TempDir(), "re-mount2")
-
-	// try mounting the same hash second time
-	os.RemoveAll(testMountDir2)
-	os.MkdirAll(testMountDir2, 0777)
-	_, err := swarmfs.Mount(bzzHash1, testMountDir2)
-	if err != nil {
-		t.Fatalf("Error mounting hash  %v", bzzHash1)
-	}
+	for _, toEncrypt := range []bool{false, true} {
```

Comment:
@gbalint I think this might be wrong.

We have a lot of cleanup code triggered with `defer` statements, which now goes inside the for loop, which initialises tests for "with" and "without" encryption.

This could very well be the reason why fuse tests are deadlocking lately.

Could you maybe try to trigger "with" and "without" encryption tests with `t.Run`, so that we don't change semantics for the cleanups triggered with `defer`-s?

Reply:
Note that this is not the only place where we have a for-loop running tests for toEncrypt -> false, true... obviously all of them should be refactored IMO.

Reply:
@nonsense ok, thanks for the info, i'm fixing this